Add updates to keyless integration guide (#343)
aptos-labs/developer-docs: apps/docusaurus/docs/guides/keyless-accounts.md

   2. Instantiate the user’s `KeylessAccount`
   3. Sign and submit transactions via the `KeylessAccount`.

> ## Example Implementaion
> 
> You can find an example app demonstrating how to do basic Keyless integration with Google in the repository below.  Follow the directions in the README to get started quickly with Keyless.  For more detailed instructions please read the rest of the integration guide.
> 
> https://github.com/aptos-labs/aptos-keyless-example/
> 

Reviewer: ```suggestion
## Example Implementaion

You can find an example app demonstrating basic Keyless integration with Google in the [aptos-keyless-example repository](https://github.com/aptos-labs/aptos-keyless-example/).  Follow the directions in the README to start with the example.  For more detailed instructions on keyless, please read the rest of this integration guide.

```

        const committedTransactionResponse = await aptos.waitForTransaction({ transactionHash: committedTxn.hash });
        ```

### 4. Error Handling

Eventually your EphemeralKeyPair will expire or the JWK used to validate the token will be rotated.  In these cases the KeylessAccount must be refreshed with a new JWT token.  To detect errors, use a try catch on sign and submit and switch on the KeylessError.type.

>   ```tsx

Reviewer: change to `ts`

>   ```tsx

Reviewer: ```suggestion
  ```ts
```

  try {
    const committedTxn = await aptos.signAndSubmitTransaction({ signer: keylessAccount, transaction });
  } catch(error) {
    if (error instanceOf KeylessError) {
      switch(error.type) { 
        case KeylessError.EPK_EXPIRED: { 
            // Handle error
            break; 
        } 
        case KeylessError.JWK_EXPIRED: { 
            // Handle error
            break; 
        } 
        default: { 
            break; 
        } 
      } 
    }
  }
  ```

## SDK Configurable Options

> The Keyless SDK provides several configurable options for Keyless Account derivation

Reviewer: ```suggestion
The Keyless SDK provides several options for Keyless Account derivation.
```

### Choice of pepper

> Developers may opt to use their own pepper service/scheme and can override use of the labs hosted pepper service.

Reviewer: ```suggestion
Developers may opt to use their own pepper service/scheme instead of the Aptos Labs hosted pepper service.
```

Example

> ```tsx

Reviewer: change to `ts`

> ```tsx

Reviewer: ```suggestion
```ts
```

const keylessAccount = await aptos.deriveKeylessAccount({
    jwt,
    ephemeralKeyPair,
    pepper: "05e2844f1a7d5d27ca57f89d6f599fb073b2352b97d4905fa1c431236e00eb",
});
```

### Claim to use for user identification

> Certain dApps may want to use 'email' instead of the default 'sub' claim to identify user.  This allows dApps to mint NFTs to a user's email even if the user has not created an account yet.  Note that this also means that if a user changes their emails with respect to the IdP, they may love access to their account.

Reviewer: ```suggestion
Certain dApps may use `email` instead of the default `sub` claim to identify users.  This allows dApps to mint NFTs to a user's email even if the user has not created an account yet.  Note if a user changes their emails with respect to the IdP, they may lose access to their account.
```

Example

> ```tsx

Reviewer: change to `ts`

> ```tsx

Reviewer: ```suggestion
```ts
```

const keylessAccount = await aptos.deriveKeylessAccount({
    jwt,
    ephemeralKeyPair,
    uidKey: "email",
});
```

### Asyncronous proof fetching

> Fetching the proof may take a few seconds and can be done in the background to allow for a more responsive user experience.  To enable asyncronous proof fetching just provide a callback to be invoked on proof fetch completion.

Reviewer: ```suggestion
Fetching a proof may take a few seconds and can be done in the background to allow for a more responsive user experience.  To enable asynchronous proof fetching, provide a callback to be invoked on proof fetch completion.
```

Example

> ```tsx

Reviewer: ```suggestion
```ts
```

const handleProofFetchStatus = async (res: ProofFetchStatus) => {
  if (res.status === 'Failed') {
    // Handle failue
    return
  }
  // Handle success
  store.persist.rehydrate();
};
const keylessAccount = await aptos.deriveKeylessAccount({
    jwt,
    ephemeralKeyPair,
    proofFetchCallback: handleProofFetchStatus,
});
```

### EphemeralKeyPair expiry time 

> By default, the expiry time of the EphemeralKeyPair is set to be maximum allowed time in the future, now + 10000000 seconds.  To avoid proofs that are long lived which can be a security risk, 
> ```tsx
> const ephemeralKeyPair = EphemeralKeyPair.generate({expiryDateSecs: 1721397500});
> ```

Reviewer: This doesn't describe the purpose of this well.  And I think it's taking some jumps assuming the user knows what these mean.  It says security risk, and it says to avoid it, but why?

```suggestion
By default, the expiry time of the EphemeralKeyPair is set to be the maximum allowed time in the future, now + 10000000 seconds.  To reduce the amount of time that a proof expires, you can override the expiration date seconds. 
```ts
const ephemeralKeyPair = EphemeralKeyPair.generate({expiryDateSecs: 1721397500});
```
```

## Pepper Service API

> The pepper service is an Aptos Labs hosted API that computes a pepper as a result of a verifiable unpredictible function (VUF).  The inputs to the VUF are the iss, uid_key (defaults to sub), uid_val, aud.

Reviewer: Okay... there are like 10 abbreviations here, and it doesn't read well.  Should we describe what a pepper is?

Pepper service makes peppers, but why?

```suggestion
The pepper service is an Aptos Labs hosted API that computes a pepper as a result of a verifiable unpredictible function (VUF).  The inputs to the VUF are the `iss`, `uid_key` (defaults to `sub`), `uid_val`, `aud`.
```

Example request
```json
{
    "jwt_b64": "eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCIsImtpZCI6InRlc3QtcnNhIn0.eyJpc3MiOiJ0ZXN0IiwiYXVkIjoidGVzdCIsInN1YiI6InRlc3QiLCJlbWFpbCI6InRlc3QiLCJlbWFpbF92ZXJpZmllZCI6dHJ1ZSwiaWF0IjoxNzE1OTI3NTI2LCJleHAiOjI3MDAwMDAwMDAsIm5vbmNlIjoiMTY1NjkyNDIwNzExODc3NTA3OTMwODk3NDIwMzcxMzUzMjY0NzY0MzQyMTI4MDgyNTM2MDA1MzEwMTE2NDQzNzAzNTMzMTkxNjIzNDEifQ.PcSnAEebQeKP2ge8hQizIazekQb_x9xS6xNqZM4cmRseisfHpXelLPGfwWsecGa6f3SZWXYeCMCtij2OXXMveYfPYTbDd8d01Ks8lHvcr5PLuRP1jvwVfLP34oKRsjjE1Dowams1xyKDmEI8qWA3IY5QILH6xIIhnnz4Qa6AvibzG_1cbXmGDaC4_zzAvL812QSzQfJ3hr45j_EAQcFOnuy_vUm7WqLVxWlWb0Vi3D5ahZe6wdlS4otcwXG8R5P94qVdZECSzsGwOyGPztUTwZuUGZV20-sxJPZmyrWNXZoNqOs3FQuAN6QfkFTcmenU1roNBZ2XTzBEGh2PDGV1ow",
    "epk": "11208fe2661376a2d938e77123d51f06a42116a0da22aaf79cf9e80527bc91bfaab8",
    "epk_blinder": "258a117208bab6b620ed4edd496fd63d327daca0d17f8d460ec2dd0bcb7a90",
    "exp_date_secs": 1725926400,
}
```

Example response
```json
{
    "pepper": "f634a051057e7c8632194c80a5c7c4d0cd10edbd8fc4505b7d2018839b0f6f",
    "address": "0xd19c3bcd94cdb6576b4a0ed958ed94805b78e1d7f4bdab3e5033bd7fb09d9bbd"
}
```

## Prover Service API

> The prover service is an Aptos Labs hosted API that computes zero-knowledge proofs.

Reviewer: proofs for what?

> The input to the prover consists of the same request to the pepper service with the addition of the pepper and the max expiry horizon.  The inputs to the prover must satisfy the circuit relation for secret witness generation.

Reviewer: Prover must satisfy circuit relation  is a weird sentence
```suggestion
The prover service takes the same request as the pepper service with two additional inputs, the pepper and the max expiry horizon. The inputs to the prover must satisfy the circuit relation for secret witness generation.
```
